CONTRIBUTING.md

```diff
@@ -47,3 +47,52 @@ pnpm changeset
   - `packages/svg-icons/src/icons/32px`
 
 - Steps 2-5 are the same as adding a new icon.
+
+## Developing a new component
+
+### Checklist:
+
+- [ ] Every component should have a GlobalCssSelector that is unique to the component. This allows the targetting of the component in the global css file.
+```
+export const GlobalIconCssSelector = "hop-icon";
+```
+
+- [ ] Every component should make sure to merge their props and ref with the context props and ref.
+```
+// with a default slot:
+[props, ref] = useContextProps({ ...props, slot: props.slot || DefaultIconListSlot }, ref, IconListContext);
+
+// without a default slot
+[props, ref] = useContextProps(props, ref, IconListContext);
+```
+- [ ] Every component should then remove the styling props from the props object and then add the other ones to the ownProps obj.
+```
+const { stylingProps, ...ownProps } = useStyledSystem(props);
+```
+
+- [ ] You can now deconstruct the ownProps object to get the props you need.
+```
+const { children, style, className, slot, ...otherProps } = ownProps;
+```
+
+- [ ] make sure to merge the classnames and styles from the props with the classnames and styles from the style system.
+Make sure to that the style from the props have higher priority than the style from the style system.
+```
+    const classNames = clsx(
+        className,
+        GlobalIconCssSelector,
```

Comment:
function names (cssModule): camelCase
constant names (GlobalIconCssSelector): PascalCase
variables(className): camelCase

Reply:
my guess is because variables can be updated, but constants should never be updated. It's a good idea to differentiate between the two. I think the old style of this used to be all caps with _ for spaces

Reply:
I'll resolve for now, but truth is, this standard is not enforced by any linting rules nor any standard across gsoft, so it's not that important